installer/mac/preinstall.sh

```diff
@@ -1,3 +1,60 @@
 #!/usr/bin/env bash
+
+# Kill the existing running instance
 sudo killall Symphony
-sudo rm -rf /Applications/Symphony.app
\ No newline at end of file
+
+compare_versions()
+{
+    # Get the installer version:
+    CURRENT_VERSION=3.1.0
+
+    # Get the currently installed version:
+    INSTALLED_VERSION=$(plutil -p /Applications/Symphony.app/Contents/Info.plist | awk '/CFBundleShortVersionString/ {print substr($3, 2, length($3)-2)}')
+
+    # If there are no versions installed, just exit the script
+    if [ -z "$INSTALLED_VERSION" -a "$INSTALLED_VERSION" != " " ]; then
+        echo "No version installed, so, exiting without version checks"
+        exit 0
+    fi
+
+    echo "This version is ${CURRENT_VERSION}"
+    echo "Installed version is ${INSTALLED_VERSION}"
+
+    # First, we replace the dots by blank spaces, like this:
+    VERSION=${CURRENT_VERSION//./ }
+    INSTALLED_VERSION=${INSTALLED_VERSION//./ }
+
+    # If you have a "v" in front of your versions, you can get rid of it like this:
+    VERSION=${VERSION//v/}
+    INSTALLED_VERSION=${INSTALLED_VERSION//v/}
+
+    # So, we just need to extract each number:
+    patch1=$(echo ${VERSION} | awk '{print $3}')
+    minor1=$(echo ${VERSION} | awk '{print $2}')
+    major1=$(echo ${VERSION} | awk '{print $1}')
+
+    patch2=$(echo ${INSTALLED_VERSION} | awk '{print $3}')
+    minor2=$(echo ${INSTALLED_VERSION} | awk '{print $2}')
+    major2=$(echo ${INSTALLED_VERSION} | awk '{print $1}')
+
+    # And now, we can simply compare the variables:
+    if [ ${major1} -lt ${major2} ]; then
+        echo "Installed version is newer than this version, exiting installation"
+        exit 1
+    fi
+
+    if [ ${minor1} -lt ${minor2} ]; then
```

Comment:
The installation would fail if the older version is `2.11.8` and the new version is `3.1.0`, as this is only comparing the minor versions

Reply:
Yes indeed, noticed that 👍 will fix and push the code.